src/twister2/specification_processor.py

```diff
@@ -269,6 +270,13 @@ def should_skip_for_pytest_harness(test_spec: YamlTestSpecification, platform: P
     return False
 
 
+def should_skip_for_spec_type_unit(test_spec: YamlTestSpecification, platform: PlatformSpecification) -> bool:
+    if test_spec.type == 'unit':
```

Comment:
Shouldn't be: `if (platform.type == 'unit') != (test_spec.type == 'unit')` ?

Reply:
I thought I added comment about it, but apparently I only explained it in the commit msg. I am not sure if we have all pieces to be able to run such tests on platform with type == 'unit'. Maybe it will work out of the box, but since I didn't have time to properly evaluate it I left it of completely. I will add a missing comment to the code